pkg/query/storeset.go

```diff
@@ -218,7 +218,10 @@ func (s *StoreSet) Update(ctx context.Context) {
 	// Record the number of occurrences of external label combinations for current store slice.
 	externalLabelOccurrencesInStores := map[string]int{}
 	for _, st := range healthyStores {
-		externalLabelOccurrencesInStores[externalLabelsFromStore(st)]++
+		if st.storeType != nil && (st.storeType.ToProto() == storepb.StoreType_QUERY ||
```

Comment:
Why are we changing this? Let's maybe focus on line 258 as you did instead (:

Reply:
Thanks! Done!